### DISCUSSION TITLE
Renovate job failing with ERROR: Disk space error - skipping

### DISCUSSION THREAD

**Discussion by simonlovesyou** [renovatebot/renovate#13204] [Request Help] [answered]
## What Renovate type, platform and version are you using?
**Type:** Renovate Hosted App
**Platform:** GitHub
**Language:** javascript/typescript
**Manager:** yarn (3.1.0)

## Describe the bug
Renovate jobs on a private repo continuously failing with caught-error status in dashboard. Root cause in the logs: 
```
DEBUG: Unknown error committing files         (branch="renovate/jest-monorepo")
{
  "err": {
    "task": {
      "commands": [
        "reset",
        "--hard"
      ],
      "format": "utf-8",
      "parser": "[function]"
    },
    "message": "error: update_ref failed for ref 'HEAD': cannot lock ref 'HEAD': Unable to create '/mnt/renovate/gh/matsmart/matsmart-web/.git/refs/heads/main.lock': No space left on device\n",
    "stack": "Error: error: update_ref failed for ref 'HEAD': cannot lock ref 'HEAD': Unable to create '/mnt/renovate/gh/matsmart/matsmart-web/.git/refs/heads/main.lock': No space left on device\n\n    at Object.action (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/plugins/error-detection.plugin.js:30:33)\n    at PluginStore.exec (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/plugins/plugin-store.js:21:33)\n    at /home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:95:45\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:93:16)\n    at GitExecutorChain.<anonymous> (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:77:46)\n    at Generator.next (<anonymous>)\n    at fulfilled (/home/ubuntu/renovateapp/node_modules/simple-git/src/lib/runners/git-executor-chain.js:5:58)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)"
  }
}
ERROR: Disk space error - skipping
```

## Relevant information

**Most recent failing job ID:** # 534107755
**Repository git clone size:** 591M
**Repository yarn installed size**: 3108M

## Have you created a minimal reproduction repository?

- [ ] This is a really small bug, it does not need a reproduction (think small typo)
- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [x] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Note:** I found similar app support discussions, but was unsure if I should post in other peoples support threads so I opted to create a new one. Please LMK if you'd like me to close this and post this in an existing thread. Thank you!

**Comment by olwe0002** [comment, marked as answer]
The error is raised in this line: https://github.com/renovatebot/renovate/blob/cf7c0c9a42aa74dbf24464695a5e450e6071f809/lib/modules/manager/npm/post-update/npm.ts#L198
According to [this](https://stackoverflow.com/questions/22475849/node-js-what-is-enospc-error-and-how-to-solve) thread, it should help to increase the `max_user_watches` to avoid this error. For k8s it can be done in [this](https://stackoverflow.com/questions/69199553/how-to-change-the-file-system-watcher-limit-in-kubernetes-fs-inotify-max-user-w) way.
